.github/workflows/main.yml

```diff
@@ -15,6 +15,19 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  run-mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
```

Comment:
Is there a reason not to use 3.12? It should provide access to newer type annotation syntax, and some faster performance checks due to performance boosts.

Reply:
Because pyqtgraph should support Python3.9+.  And I use Python3.9 for development to avoid backward compatibility issue.

for example, 
```python
a: int | str = 10
```
is OK in python3.10, but raise `TypeError` in 3.9

So using lower version's python to check the type makes sense.
Let me know your option, if you want, I can update it to 3.10 :)

Reply:
oh right, yeah probably should check on the oldest supported version...